### PR TITLE
MAINT: 1.7.x version pins ("backport")

### PR DESCRIPTION
**Pull request by tylerjereddy**
@rgommers Chances are this isn't quite right yet--it mostly borrows from the changes in the previous release branch, so maybe some versions can be slightly higher..

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,10 +9,10 @@
 
 [build-system]
 requires = [
-    "wheel",
+    "wheel<0.37.0",
```
Comment (rgommers) [inline pyproject.toml:12]: Current version is `0.36.2`, so this is correct.

File: pyproject.toml [hunk 2]
```diff
@@ -9,10 +9,10 @@
...
     "setuptools",
-    "Cython>=0.29.18",
-    "pybind11>=2.4.3",
+    "Cython>=0.29.18,<3.0",
+    "pybind11>=2.4.3,<2.7.0",
```
Comment (rgommers) [inline pyproject.toml:15]: Current version is `2.6.2`, so this is correct.

File: pyproject.toml [hunk 3]
```diff
@@ -9,10 +9,10 @@
 
 [build-system]
 requires = [
-    "wheel",
+    "wheel<0.37.0",
     "setuptools",
-    "Cython>=0.29.18",
-    "pybind11>=2.4.3",
+    "Cython>=0.29.18,<3.0",
```
Comment (rgommers) [inline pyproject.toml:14]: Should be correct indeed.

File: scipy/__init__.py [hunk 4]
```diff
@@ -139,7 +139,7 @@ def __dir__():
     # In maintenance branch, change to np_maxversion N+3 if numpy is at N
     # See setup.py for more details
     np_minversion = '1.16.5'
-    np_maxversion = '9.9.99'
+    np_maxversion = '1.23.0'
```
Comment (rgommers) [inline scipy/__init__.py:142]: This is right; if `1.21.0` is released before our last 1.7.0 RC then we could consider bumping this to `1.24.0`

**Review by rgommers — COMMENTED** [review]
Changes look good, but we'll need some upper bounds for other build dependencies (`setuptools`, `pythran`) as well. I'll have a look at updating those.

**Comment by tupui** [comment]
In the `pyproject.toml` shouldn't we remove the references to Python 3.6?

**Comment by rgommers** [comment]
> In the `pyproject.toml` shouldn't we remove the references to Python 3.6?

Yes, we can indeed. We may also be missing some `python_version` specifiers for 3.8-3.9 for `aarch64`, as well as AIX specifiers. I'll compare with `oldest-supported-numpy`.

File: pyproject.toml [hunk 5]
```diff
@@ -9,11 +9,11 @@
...
-    "wheel",
-    "setuptools",
-    "Cython>=0.29.18",
-    "pybind11>=2.4.3",
-    "pythran>=0.9.11",
+    "wheel<0.37.0",
+    "setuptools<58.0.0",
+    "Cython>=0.29.18,<3.0",
+    "pybind11>=2.4.3,<2.7.0",
+    "pythran==0.9.11",
```
Comment (rgommers) [inline pyproject.toml:16]: Pinning with `==` to ensure we get that exact version, given our limited experience with it I don't want to say `>=` which can give not-yet-released bugfix versions.

**Comment by rgommers** [comment]
That cleans it up a bit. We only need three sets of requirements: `aarch64`, `PyPy` and the default ones.

**Comment by tylerjereddy** [comment]
Thanks Ralf and Pamphile, I'll put this backport in

**Pull request merged**